File: JAVA/Point.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package JavaStarterCode;


/**
 *
 * @author singh
 */
public class Point {
    private Long x,y;
    
    public Point() {}

    /*
    Point: Initialise with x and y coordinates of Elliptic curve point
     */
    public Point(Long x, Long y)
    {
        this.x = x;
        this.y = y;
    }

    /*
    getX: Return x coordinate of Elliptic curve point
     */
    public Long getX()
    {
        return x;
    }

    /*
    getY: Return y coordinate of Elliptic curve point
     */
    public Long getY()
    {
        return y;
    }

    /*
    setX: Store x coordinate of Elliptic curve point
     */
    public void setX(Long x)
    {
        this.x = x;
    }

    /*
    setY: Store y coordinate of Elliptic curve point
     */
    public void setY(Long y)
    {
        this.y = y;
    }
}

File: Python/Point.py
class Point:
	# __init__: Initialise with x and y coordinates of Elliptic curve point
	def __init__(self, x, y):
		self.x = x
		self.y = y

	# __str__: Allows Point objects to be printed intuitively
	def __str__(self):
		return "(" + str(self.x) + "," + str(self.y) + ")"

	# __eq__: Allows Point objects to be compared
	def __eq__(self, other):
		return self.getX() == other.getX() and self.getY() == other.getY()

	# getX: Return x coordinate of Elliptic curve point
	def getX(self):
		return self.x

	# getY: Return y coordinate of Elliptic curve point
	def getY(self):
		return self.y
